packages/core/server/internal-base/src/plugins.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * Dedicated type for plugin name/id that is supposed to make Map/Set/Arrays
+ * that use it as a key or value more obvious.
+ *
+ * @public
+ */
+export type PluginName = string;
+
+/** @public */
+export type PluginOpaqueId = symbol;
```

Comment:
This will move to `@kbn/core-common-internal-base` in a follow-up as it's also used on the client-side